nixos/modules/services/security/step-ca.nix

```diff
@@ -0,0 +1,113 @@
+{ config, lib, pkgs, ... }:
+
+let cfg = config.services.step-ca;
+
+in {
+  options = {
+    services.step-ca = {
+      enable = lib.mkEnableOption "Step CA PKI Server.";
+      stateDir = lib.mkOption {
+        type = lib.types.str;
+        default = "step-ca";
+        description = "Directory name in /var/lib/ to store Step CA Service data";
+      };
+      name = lib.mkOption {
+        type = lib.types.str;
+        default = "SmallStep";
+        description = "Name of your CA.";
+      };
+      hostname = lib.mkOption {
+        type = lib.types.str;
+        default = "localhost";
+        description = "Hostname/FQDN of your CA.";
+      };
+      address = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+        example = "0.0.0.0";
+        description = ''Address to listen at, defaults to all interfaces.'';
+      };
+      port = lib.mkOption {
+        type = lib.types.port;
+        default = 8443;
+        description = "Port to listen on.";
+      };
+      passwordFile = lib.mkOption {
+        type = lib.types.path;
+        default = null; # TODO test this
+        description = "Path to file containing password of your first provisioner.";
+      };
+      provisioner = lib.mkOption {
+        type = lib.types.str;
+        default = "you@smallstep.com";
+        description = "The name of the first provisioner.";
+      };
+      acmeProvisioner = lib.mkOption {
+        type = lib.types.nullOr lib.types.str;
+        default = null;
+        example = "acme";
+        description = ''Optionally create an acme provider. The resulting acme url is https://{hostname}:{port}/acme/{acmeProvisioner}/directory'';
+      };
+      rootFile = lib.mkOption {
+        type = lib.types.nullOr lib.types.path;
+        default = null;
+        description = ''The path of an existing PEM file to be used as the root certificate authority. You can generate one using e.g `step certificate create "MyCompany" root-ca.crt root-ca.key --profile root-ca --no-password --insecure` from the `step-cli` package. Only use this for development as the ca key is not protected with a password. It seems like step-ca ignores the passwordFile parameter so you can't use a key with a password. This should probably be reported as a bug. https://github.com/smallstep/cli/issues/220'';
+      };
+      keyFile = lib.mkOption {
+        type = lib.types.nullOr lib.types.path;
+        default = null;
+        description = "The path of an existing key file of the root certificate authority.";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.step-ca = {
+      wantedBy = [ "multi-user.target" ]; # todo fixme this should be like acme
+      after = [ "network.target" ];
+      path = [ pkgs.coreutils pkgs.step-ca pkgs.step-cli ];
```

Comment:
I believe you can safely remove `coreutils` from `path` if it's only being used for `ExecReload`. Since the full path is being passed to `ExecReload`, it doesn't need to look at `PATH` to find the command.